compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -17300,9 +17318,11 @@ private STNode parseStmtStartsWithTupleTypeOrExprRhs(STNode annots, STNode tuple
 
     private STNode parseAsTupleTypeDesc(STNode annots, STNode openBracket, List<STNode> memberList, STNode member,
                                         boolean isRoot) {
-        memberList = getTypeDescList(memberList);
+        memberList = getTupleMemberList(memberList);
+        startContext(ParserRuleContext.TUPLE_MEMBERS);
         STNode tupleTypeMembers = parseTupleTypeMembers(member, memberList);
         STNode closeBracket = parseCloseBracket();
+        endContext();
```

Comment:
At this point `stmt-start-bracketed-list` ctx due to ambiguity. When we reach here we know definite it is tuple. So we can switch it to the tuple.
```suggestion
        switchContext(ParserRuleContext.TUPLE_MEMBERS);
        STNode tupleTypeMembers = parseTupleTypeMembers(member, memberList);
        STNode closeBracket = parseCloseBracket();
```

Reply:
My bad. you are right here.